docs/variables.rst

```diff
@@ -0,0 +1,131 @@
+=========
+Variables
+=========
+
+Variables can be parsed into a rulebook using one or both of these command line arguments::
+
+    --vars VARS           Variables file
+    --env-vars ENV_VARS   Comma separated list of variables to import from the environment
+
+To import variables from a JSON or YAML formatted file, the `--vars` argument should be used.
+To import environment variables the `--env-vars` argument should be used. It is possible to specify
+multiple environment variables separated by a comma.
+
+.. important::
+    Variables that are supplied using the `--vars` argument will have their data type preserved, but environment
+    variables parsed via `--env-vars` are always treated as strings. This is important to consider when parsing
+    variables into your rulebook.
+
+.. note::
+    In cases where the same variable is supplied via `--env-vars` and `--vars`, the one from
+    `--env-vars` will always take precedence.
+
+
+Accessing variables in your rulebook
+------------------------------------
+
+The are two important considerations to take into account when accessing variables in a rulebook:
+
+    1. In order to use variables in conditions, the variables supplied at runtime will be placed into a namespace
+    called **vars**. You must use the **vars.** prefix followed by the variable name to use variables in conditions.
+
+    2. Jinja substitution should be used when accessing variables in any other part of the rulebook, such as in
+    the source configuration or in actions.
+
+Let's take the following example rulebook:
+
+    .. code-block:: yaml
+
+        ---
+        - name: Testing vars
+          hosts: all
+          sources:
+            - ansible.eda.range:
+                limit: "{{ src_range_limit }}"
+          rules:
+            - name: Say Hello
+              condition: event.i == vars.match_this_int
+              action:
+                echo:
+                  message: "Hi, I'm {{ MY_NAME }}."
+
+In this rulebook we use three different variables. One, **MY_NAME**, we will supply using an environment variable.
+The other two will be supplied in a file called `vars.yml` which looks like this:
+
+    .. code-block:: yaml
+
+        ---
+        src_range_limit: 4
+        match_this_int: 2
+
+To run this rulebook we will need to use both `--vars` and `--env-vars` to parse the variables into the rulebook. Therefore,
+we would execute the rulebook as follows:
+
+    .. code-block:: console
+
+        $ MY_NAME="Bob" ansible-rulebook -i inventory.yml --rulebook hello_there.yml --vars vars.yml --env-vars MY_NAME
+
+Which would return the following output:
+
+    .. code-block:: console
+
+        2023-02-09 13:37:73.311337 : Hi, I'm Bob
+
+.. note::
+    - Quotation marks around Jinja substituted variables are **mandatory** when used as the first element in a value.
```

Comment:
I don't understand the last part of this sentence. Quotation marks around Jinja expressions are mandatory *always*. Conditions do not use jinja. Maybe it's a matter to remark that difference.

Reply:
An example will help.

Reply:
@Alex-Izquierdo it is best practice, but it is not mandatory. This is true in Ansible and ansible-rulebook. For example, the following will work:
`message: The action is {{ msg }}`

But this will fail:
`message: {{ msg }} is the action`

It is mentioned in the Ansible [docs](https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html#gotchas) but since its a common 'gotcha' that folks run into I think we should mention it in our docs as well. Ansible has a clear error message when you hit do it wrong, but ansible-rulebook does not.

I can add an example.

Reply:
Or we change the note to recommend always enclosing values that contain Jinja substituted variables in quotation marks.

Reply:
IMO better and failsafe saying use always quotation marks